WeatherMonitoringAndReportingService/DataSourceProcessor/JSONFileProcessor.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using WeatherMonitoringAndReportingService.AppSettings;
+using WeatherMonitoringAndReportingService.Config;
+
+namespace WeatherMonitoringAndReportingService.DataSourceProcessor;
+
+public class JSONFileProcessor : IDataSourceProcessor
+{
+    public void Add(string name, WeatherConfigurationModel data, string path)
```

Comment:
According to Interface declaration

```suggestion
    public void Add(string name, WeatherConfigurationModel data, string? path)
```